### PR TITLE
feat: add binary plugins

### PR DESCRIPTION
**Pull request by pmuir**
Binary plugins in jx behave exactly like binary plugins in kubectl, see 
https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -162,11 +164,13 @@ require (
 	k8s.io/api v0.0.0-20180628040859-072894a440bd
 	k8s.io/apiextensions-apiserver v0.0.0-20180621085152-bbc52469f98b
 	k8s.io/apimachinery v0.0.0-20180621070125-103fd098999d
+	k8s.io/cli-runtime v0.0.0-20181026155151-1ee5ba10d7e3 // indirect
 	k8s.io/client-go v8.0.0+incompatible
 	k8s.io/code-generator v0.0.0-20181017053441-8c97d6ab64da
 	k8s.io/gengo v0.0.0-20180718083919-906d99f89cd6
 	k8s.io/helm v2.7.2+incompatible
 	k8s.io/kube-openapi v0.0.0-20180719232738-d8ea2fe547a4
+	k8s.io/kubernetes v1.12.2 // indirect
```
Comment (ccojocar) [inline go.mod]: Probably we want to keep this dependency to v1.11.3?

File: pkg/jx/cmd/plugin.go [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+func (o *PluginListOptions) Complete() error {
+	o.Verifier = &CommandOverrideVerifier{
+		root:        o.Cmd.Root(),
+		seenPlugins: make(map[string]string, 0),
+	}
+	return nil
+}
+
+func (o *PluginListOptions) Run() error {
```
Comment (ccojocar) [inline pkg/jx/cmd/plugin.go]: It would be nice to have a test for this.

**Comment by pmuir** [comment]
/hold

Putting this explicitly on hold as I need to come back to this in a couple of days :-D

File: go.mod (hunk 1 above)
Comment (pmuir) [reply go.mod]: Fixed

**Pull request merged**